src/Concerns/HasRelations.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace Swis\JsonApi\Client\Concerns;
+
+use Illuminate\Support\Str;
+use Swis\JsonApi\Client\Collection;
+use Swis\JsonApi\Client\Interfaces\DataInterface;
+use Swis\JsonApi\Client\Interfaces\ManyRelationInterface;
+use Swis\JsonApi\Client\Interfaces\OneRelationInterface;
+use Swis\JsonApi\Client\Links;
+use Swis\JsonApi\Client\Meta;
+use Swis\JsonApi\Client\Relations\HasManyRelation;
+use Swis\JsonApi\Client\Relations\HasOneRelation;
+use Swis\JsonApi\Client\Relations\MorphToManyRelation;
+use Swis\JsonApi\Client\Relations\MorphToRelation;
+
+trait HasRelations
+{
+    /**
+     * @var \Swis\JsonApi\Client\Interfaces\OneRelationInterface[]|\Swis\JsonApi\Client\Interfaces\ManyRelationInterface[]
+     */
+    protected $relations = [];
+
+    /**
+     * Create a singular relation to another item.
+     *
+     * @param string      $itemClass
+     * @param string|null $name
+     *
+     * @return \Swis\JsonApi\Client\Relations\HasOneRelation
+     */
+    public function hasOne(string $itemClass, string $name = null): OneRelationInterface
+    {
+        $name = $name ?: Str::snake(debug_backtrace()[1]['function']);
```

Comment:
You could refactor `Str::snake(debug_backtrace()[1]['function']);` into a method for readability and DRY reasons